test/src/unit-cppapi-deletes.cc

```diff
@@ -418,8 +418,14 @@ TEST_CASE_METHOD(
   // Write another fragment that will not be affected by the condition.
   write_sparse({1}, {4}, {4}, 5);
 
+  // Test read for both refactored and legacy.
+  bool legacy = GENERATE(true, false);
+  if (legacy) {
+    set_legacy();
+  }
+
   // Reading before the delete condition timestamp.
-  uint64_t buffer_size = 4;
+  uint64_t buffer_size = legacy ? 100 : 4;
```

Comment:
Why is the buffer size so much larger for legacy?

Reply:
The legacy reader does result size estimation versus the user buffers... If it thinks the data won't fit, it will partition. This size estimation doesn't take into account query condition or delete condition so if we give the reader buffers that will just fit the data, it will likely partition, which we want to avoid for this test.